esmvaltool/cmorizers/obs/utilities.py

```diff
@@ -155,7 +155,7 @@ def save_variable(cube, var, outdir, attrs, **kwargs):
     except iris.exceptions.CoordinateNotFoundError:
         time_suffix = None
     else:
-        if len(time.points) == 1:
+        if len(time.points) == 1 and cube.attributes.get('mip') != 'Amon':
```

Comment:
@mwjury could you please explain why specifically mip: Amon? It could be a lot others

Reply:
So the original code has an exception assuming that if there is only one time point it is automatically data for the whole year and sets the time_suffix = '-'.join([year + '01', year + '12']). #1991 also builds monthly data from daily data stored in monthly files. I know it would be more sane to check if it is yearly data in an other way, but I didn't want this to cause problems for existing cmorizers.
The existing version produces problems for all data that have len(time.points) == 1 that are not yearly means.

Reply:
good point but let's not fix an issue by introducing another one :grin: Let me just suggest a code change via the github editor